### ISSUE TITLE
Unsafe port 6000 in Firefox and Chrome

### ISSUE THREAD

**Issue by copystring** [engageub/InternetIncome#109]
Hi,

Firefox and Chrome are blocking port 6000 because of security reasons. Please change the starting port of custom Firefox from 6000 to 6001 or anything else like 8000. 8000 Seems to be the better option if you ask me.
See this list: https://neo4j.com/developer/kb/list-of-restricted-ports-in-browsers/

**Comment by copystring** [comment]
Maybe 9000 would be better because port 8000 is used by portainer by default.

**Comment by engageub** [comment]
Hello,
The ports are assigned based on availability.
If port 6000 is not available, the script looks for next available port
before assigning.
Could you please let me know if something was not
working?
Thank you

**Comment by copystring** [comment]
Firefox and Chrome don't like port 6000. Both browsers block it. This has nothing to do with available ports.
![image](https://github.com/engageub/InternetIncome/assets/1298861/ef3ea5c0-f343-4ca9-b24a-a1173872a479)

**Comment by copystring** [comment]
`ERR_UNSAFE_PORT`

**Comment by engageub** [comment]
I have not seen any issue with my VPS on those ports. In addition to that,
you have confirmed it to be working earlier.
Could you please let me know if you have installed any other application.

Thank you

**Comment by engageub** [comment]
Hello,
I used serveo.net to access it globally. Can you try the following with
ports mentioned in text file.

  ssh -R 80:localhost:6000 serveo.net

Thank you

**Comment by copystring** [comment]
I tried this. When I want to open the URL served by serveo.net it gives me error 502

**Comment by copystring** [comment]
`ssh -R 80:localhost:6001 serveo.net` works perfectly fine. `ssh -R 80:localhost:6000 serveo.net` does not and errors like this:
![image](https://github.com/engageub/InternetIncome/assets/1298861/5e536680-7d0a-44f3-b3b9-0db9a5eba646)
Also in port 6000 is usually X11 running.

**Comment by copystring** [comment]
![image](https://github.com/engageub/InternetIncome/assets/1298861/2d8552cc-159c-4e51-b10e-15a0b160f422)

**Comment by engageub** [comment]
Strange. Only after viewing VNC, changes were pushed to test branch.
You may modify the initial port of custom firefox in
the code to access it on your host.
I will login to my host sometime later and try to replicate it.
Could you please specify your PC config whether it is amd64 or arm64.

**Comment by copystring** [comment]
> You may modify the initial port of custom firefox in the code to access it on your host.

Yes, when I change the port from 6000 to 8000 it works fine, but 8000 isn't a great choice because port 8000 is used by portainer.


> Could you please specify your PC config whether it is amd64 or arm64.

I'm running a amd64 system

**Comment by engageub** [comment]
Do you see url with port 6000 in custom_firefox.txt
Port checking may also have failed while assigning.

**Comment by engageub** [comment]
You may change to any port number. It checks for next available port before
assigning.

**Comment by copystring** [comment]
> Do you see url with port 6000 in custom_firefox.txt

Yes, port 6000 is listed and assigned properly. But like I said above, it is not possible to access port 6000 because browsers block accessing the port even though it is assigned correctly, it cannot be opened by browsers.

**Comment by engageub** [comment]
I could access port 6000 on both arm64 and amd64. May be you have something running on port 6000 and the script was not able to detect the open port.

I would like you to run the following commands and share the output. The following command is being used in script which checks if port is available before assigning. Port 6000 is already added below. Please run the command and let me know the output.

```
{ timeout 1 bash -c "echo > /dev/tcp/localhost/6000" > /dev/null; } 2>/dev/null
    if [ $? -eq 0 ]; then
      echo "Port is not available"
    else
      echo "Port is available to be used"
    fi

```


Please also run the following commands and share the output. 
```
sudo lsof -i:6000
```

The following is used in main branch. Please also run this and share the output. 

```
nc -z localhost 6000 > /dev/null 2>&1
if [ $? -eq 0 ]; then
  echo "Port is not available"
else
  echo "Port is available to be used"
fi
```

**Comment by copystring** [comment]
Here are the outputs after running your script. If I don't run your script, port 6000 is available.

```
# { timeout 1 bash -c "echo > /dev/tcp/localhost/6000" > /dev/null; } 2>/dev/null
    if [ $? -eq 0 ]; then
      echo "Port is not available"
    else
      echo "Port is available to be used"
    fi
Port is not available
```


```
# sudo lsof -i:6000
COMMAND       PID USER   FD   TYPE   DEVICE SIZE/OFF NODE NAME
docker-pr 3948355 root    4u  IPv4 39095564      0t0  TCP *:x11 (LISTEN)
docker-pr 3948363 root    4u  IPv6 39096504      0t0  TCP *:x11 (LISTEN)
```

```
# nc -z localhost 6000 > /dev/null 2>&1
if [ $? -eq 0 ]; then
  echo "Port is not available"
else
  echo "Port is available to be used"
fi
Port is not available
```

I know what you are trying to do, but you are on the wrong track.
Firefox and Chrome are not able to access any website with port 6000.
It does not matter if your script is running or not.
Try open google.com:6000, github.com:6000, localhost:6000. It is not possible, because port 6000 is considered as unsafe.
See the error message `ERR_UNSAFE_PORT`.

![image](https://github.com/engageub/InternetIncome/assets/1298861/eae6a728-eb38-410e-b79f-8e25203d154a)
![image](https://github.com/engageub/InternetIncome/assets/1298861/e370613e-920c-475f-af74-b13d190d8042)
![image](https://github.com/engageub/InternetIncome/assets/1298861/aeb9c38f-2016-4858-a3e6-1292a28487c9)

**Comment by engageub** [comment]
Hello,
May I know why you are trying to access port 6000 on github. It is no where related to the script.
The output clearly mentions that port 6000 is not available. So it will assign port next available port 6001 instead.
I wonder how port 6000 was assigned. You have X11 running on port 6000 and the script clearly indicates port 6000 is not available and does not assign the port.
In the console output after running the command do you see port 6000 or port 6001?
It should show http://127.0.0.1:6001 instead.

Thank you

**Comment by copystring** [comment]
> May I know why you are trying to access port 6000 on github. It is no where related to the script.

For testing to show you the problem that browsers cannot access port 6000 on any website. It does not matter if it is localhost or github.com or any website. Port 6000 cannot be used.



> The output clearly mentions that port 6000 is not available. So it will assign port next available port 6001 instead.

Yes, because the docker container is running on port 6000.
![image](https://github.com/engageub/InternetIncome/assets/1298861/df5b6aab-d22d-4217-a7a2-0d8c002b0e7d)
I do not have X11 running on 6000
```
# netstat -ltnp | grep -w ':6000'
tcp6       0      0 :::6000                 :::*                    LISTEN      4030422/docker-prox
```



> In the console output after running the command do you see port 6000 or port 6001?
> It should show http://127.0.0.1:6001 instead.

The console output shows ports 6000. http://127.0.0.1:6000

**Comment by engageub** [comment]
This one does not show X11.  This looks like a different host than the one mentioned earlier. Port 6000 is assigned to docker container as seen from netstat command. 
Are you sure both the hosts are same?

**Comment by copystring** [comment]
Yes, it is the same host.
It shows X11 because the port is labeled X11. That does not mean that X11 is running. If any process is running on port 6000 it will show X11.

![image](https://github.com/engageub/InternetIncome/assets/1298861/6a388fbf-d38e-453d-a8e2-f475beee91c0)

**Comment by engageub** [comment]
I too see the same output, but I am able to access firefox using serveo on port 6000.

```
ubuntu@ubuntu:~/customBrowser/InternetIncome-test$ sudo netstat -ltnp | grep -w ':6000'
tcp6       0      0 :::6000                 :::*                    LISTEN      1174223/docker-prox
ubuntu@ubuntu:~/customBrowser/InternetIncome-test$ sudo lsof -i:6000
COMMAND       PID USER   FD   TYPE    DEVICE SIZE/OFF NODE NAME
docker-pr 1174218 root    4u  IPv4 190887176      0t0  TCP *:x11 (LISTEN)
docker-pr 1174223 root    4u  IPv6 190887181      0t0  TCP *:x11 (LISTEN)

```

```
ssh -R 80:localhost:6000 serveo.net

```
Thank you

**Comment by copystring** [comment]
![image](https://github.com/engageub/InternetIncome/assets/1298861/845befbc-3b62-4909-9944-8b05ea852502)

**Comment by engageub** [comment]
Has firefox stopped on that port 6000 due to high memory utilization?

Thank you

**Comment by copystring** [comment]
I will do --delete and then --start again.

**Comment by copystring** [comment]
> Delete the backup also as it will pick up the same data and may consume the same memory as it was earlier.

Do you mean the .zip files?

**Comment by engageub** [comment]
> > Delete the backup also as it will pick up the same data and may consume the same memory as it was earlier.
> 
> Do you mean the .zip files?

No, You have to use --deleteBackup to delete the custom data folders.

**Comment by engageub** [comment]
If you don't wish to delete and find out the reason. You have to enable logs in properties.conf and look into the first container logs.

**Comment by copystring** [comment]
I did --delete and --deleteBackup then --start
Here is the log of first custom Firefox container:

```
[cont-env    ] GTK2_RC_FILES: loading...
[cont-env    ] GTK_THEME: executing...
[cont-env    ] GTK_THEME: terminated successfully.
[cont-env    ] GTK_THEME: loading...
[cont-env    ] HOME: loading...
[cont-env    ] QT_STYLE_OVERRIDE: executing...
[cont-env    ] QT_STYLE_OVERRIDE: terminated successfully.
[cont-env    ] QT_STYLE_OVERRIDE: loading...
[cont-env    ] TAKE_CONFIG_OWNERSHIP: loading...
[cont-env    ] XDG_CACHE_HOME: loading...
[cont-env    ] XDG_CONFIG_HOME: loading...
[cont-env    ] XDG_DATA_HOME: loading...
[cont-env    ] XDG_RUNTIME_DIR: loading...
[cont-env    ] XDG_STATE_HOME: loading...
[cont-env    ] container environment variables initialized.
[cont-secrets] loading container secrets...
[cont-secrets] container secrets loaded.
[cont-init   ] executing container initialization scripts...
[cont-init   ] 10-certs.sh: executing...
[cont-init   ] 10-certs.sh: terminated successfully.
[cont-init   ] 10-check-app-niceness.sh: executing...
[cont-init   ] 10-check-app-niceness.sh: terminated successfully.
[cont-init   ] 10-clean-logmonitor-states.sh: executing...
[cont-init   ] 10-clean-logmonitor-states.sh: terminated successfully.
[cont-init   ] 10-clean-tmp-dir.sh: executing...
[cont-init   ] 10-clean-tmp-dir.sh: terminated successfully.
[cont-init   ] 10-fontconfig-cache-dir.sh: executing...
[cont-init   ] 10-fontconfig-cache-dir.sh: terminated successfully.
[cont-init   ] 10-init-users.sh: executing...
[cont-init   ] 10-init-users.sh: terminated successfully.
[cont-init   ] 10-nginx.sh: executing...
[cont-init   ] 10-nginx.sh: terminated successfully.
[cont-init   ] 10-openbox.sh: executing...
[cont-init   ] 10-openbox.sh: terminated successfully.
[cont-init   ] 10-pkgs-mirror.sh: executing...
[cont-init   ] 10-pkgs-mirror.sh: terminated successfully.
[cont-init   ] 10-set-tmp-dir-perms.sh: executing...
[cont-init   ] 10-set-tmp-dir-perms.sh: terminated successfully.
[cont-init   ] 10-vnc-password.sh: executing...
[cont-init   ] 10-vnc-password.sh: terminated successfully.
[cont-init   ] 10-web-data.sh: executing...
[cont-init   ] 10-web-data.sh: terminated successfully.
[cont-init   ] 10-x11-unix.sh: executing...
[cont-init   ] 10-x11-unix.sh: terminated successfully.
[cont-init   ] 10-xdg-runtime-dir.sh: executing...
[cont-init   ] 10-xdg-runtime-dir.sh: terminated successfully.
[cont-init   ] 15-cjk-font.sh: executing...
[cont-init   ] 15-cjk-font.sh: terminated successfully.
[cont-init   ] 15-install-pkgs.sh: executing...
[cont-init   ] 15-install-pkgs.sh: terminated successfully.
[cont-init   ] 55-check-snd.sh: executing...
[cont-init   ] 55-check-snd.sh: sound not supported: device /dev/snd not exposed to the container.
[cont-init   ] 55-check-snd.sh: terminated successfully.
[cont-init   ] 55-firefox.sh: executing...
[cont-init   ] 55-firefox.sh: terminated successfully.
[cont-init   ] 56-firefox-set-prefs-from-env.sh: executing...
[cont-init   ] 56-firefox-set-prefs-from-env.sh: terminated successfully.
[cont-init   ] 85-take-config-ownership.sh: executing...
[cont-init   ] 85-take-config-ownership.sh: terminated successfully.
[cont-init   ] 89-info.sh: executing...
    ╭――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╮
    │                                                                      │
    │ Application:           Firefox                                       │
    │ Application Version:   120.0.1-r0                                    │
    │ Docker Image Version:  24.01.1                                       │
    │ Docker Image Platform: linux/amd64                                   │
    │                                                                      │
    ╰――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╯
[cont-init   ] 89-info.sh: terminated successfully.
[cont-init   ] all container initialization scripts executed.
[init        ] giving control to process supervisor.
[supervisor  ] loading services...
[supervisor  ] loading service 'default'...
[supervisor  ] loading service 'app'...
[supervisor  ] loading service 'gui'...
[supervisor  ] loading service 'xvnc'...
[supervisor  ] loading service 'nginx'...
[supervisor  ] loading service 'certsmonitor'...
[supervisor  ] service 'certsmonitor' is disabled.
[supervisor  ] loading service 'openbox'...
[supervisor  ] loading service 'logrotate'...
[supervisor  ] loading service 'logmonitor'...
[supervisor  ] service 'logmonitor' is disabled.
[supervisor  ] all services loaded.
[supervisor  ] starting services...
[supervisor  ] starting service 'xvnc'...
[xvnc        ] Xvnc TigerVNC 1.13.1 - built Dec 21 2023 00:53:33
[xvnc        ] Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
[xvnc        ] See https://www.tigervnc.org for information on TigerVNC.
[xvnc        ] Underlying X server release 12014000
[xvnc        ] Sun Apr  7 20:50:51 2024
[xvnc        ]  vncext:      VNC extension running!
[xvnc        ]  vncext:      Listening for VNC connections on /tmp/vnc.sock (mode 0660)
[xvnc        ]  vncext:      created VNC server for screen 0
[supervisor  ] starting service 'nginx'...
[nginx       ] Listening for HTTP connections on port 5911.
[supervisor  ] starting service 'openbox'...
[supervisor  ] starting service 'app'...
[app         ] Mozilla Firefox 120.0.1
[supervisor  ] all services started.
```

Here is the log first tun2socks:

```
time="2024-04-07T20:50:41Z" level=info msg="[DIALER] set fwmark: 0x22b"
time="2024-04-07T20:50:41Z" level=info msg="[STACK] tun://tun0 <-> socks5://176.103.231.160:50101"
time="2024-04-07T20:50:54Z" level=info msg="[TCP] 198.18.0.1:60538 <-> 34.107.221.82:80"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:60548 <-> 34.107.221.82:80"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:53840 <-> 34.160.144.191:443"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:44530 <-> 34.107.243.93:443"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:56566 <-> 34.120.208.123:443"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:51560 <-> 192.229.221.95:80"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:56574 <-> 34.120.208.123:443"
time="2024-04-07T20:50:55Z" level=info msg="[TCP] 198.18.0.1:49884 <-> 184.24.77.52:80"
time="2024-04-07T20:50:56Z" level=info msg="[TCP] 198.18.0.1:44542 <-> 34.107.243.93:443"
time="2024-04-07T20:51:03Z" level=info msg="[TCP] 198.18.0.1:59268 <-> 172.64.140.7:443"
time="2024-04-07T20:51:03Z" level=info msg="[TCP] 198.18.0.1:52430 <-> 208.95.112.1:80"
time="2024-04-07T20:51:03Z" level=info msg="[TCP] 198.18.0.1:59284 <-> 172.64.140.7:443"
time="2024-04-07T20:51:04Z" level=info msg="[TCP] 198.18.0.1:46670 <-> 172.64.141.7:443"
time="2024-04-07T20:51:04Z" level=info msg="[TCP] 198.18.0.1:35752 <-> 80.240.22.34:7070"
time="2024-04-07T20:51:05Z" level=info msg="[TCP] 198.18.0.1:59288 <-> 172.64.140.7:443"
time="2024-04-07T20:51:07Z" level=info msg="[TCP] 198.18.0.1:54238 <-> 80.240.22.34:7072"
time="2024-04-07T20:51:07Z" level=info msg="[TCP] 198.18.0.1:59680 <-> 136.243.150.101:443"
time="2024-04-07T20:51:07Z" level=info msg="[TCP] 198.18.0.1:38968 <-> 172.64.141.7:80"
time="2024-04-07T20:51:09Z" level=info msg="[TCP] 198.18.0.1:46268 <-> 168.119.91.41:443"
time="2024-04-07T20:51:09Z" level=info msg="[TCP] 198.18.0.1:60330 <-> 46.4.87.83:769"
time="2024-04-07T20:51:12Z" level=info msg="[TCP] 198.18.0.1:35416 <-> 104.26.11.40:443"
time="2024-04-07T20:51:12Z" level=info msg="[TCP] 198.18.0.1:55050 <-> 104.26.11.40:8443"
time="2024-04-07T20:51:13Z" level=info msg="[TCP] 198.18.0.1:46578 <-> 142.250.185.228:443"
time="2024-04-07T20:51:21Z" level=info msg="[TCP] 198.18.0.1:54384 <-> 93.189.62.83:443"
time="2024-04-07T20:51:27Z" level=info msg="[TCP] 198.18.0.1:46676 <-> 104.26.12.49:443"
time="2024-04-07T20:51:28Z" level=info msg="[TCP] 198.18.0.1:47204 <-> 104.26.12.49:443"
time="2024-04-07T20:51:29Z" level=info msg="[TCP] 198.18.0.1:47212 <-> 104.26.12.49:443"
time="2024-04-07T20:51:30Z" level=info msg="[TCP] 198.18.0.1:37464 <-> 169.150.247.39:443"
time="2024-04-07T20:51:30Z" level=info msg="[TCP] 198.18.0.1:47222 <-> 104.26.12.49:443"
time="2024-04-07T20:51:33Z" level=info msg="[TCP] 198.18.0.1:57838 <-> 18.167.66.16:8080"
time="2024-04-07T20:51:34Z" level=info msg="[TCP] 198.18.0.1:57844 <-> 18.167.66.16:8080"
time="2024-04-07T20:51:38Z" level=info msg="[TCP] 198.18.0.1:57918 <-> 104.26.13.49:443"
time="2024-04-07T20:51:39Z" level=info msg="[TCP] 198.18.0.1:40524 <-> 18.167.66.16:8080"
time="2024-04-07T20:51:40Z" level=info msg="[TCP] 198.18.0.1:43956 <-> 162.55.131.196:443"
time="2024-04-07T20:51:40Z" level=info msg="[TCP] 198.18.0.1:53548 <-> 193.228.196.69:443"
time="2024-04-07T20:51:40Z" level=info msg="[TCP] 198.18.0.1:40526 <-> 18.167.66.16:8080"
time="2024-04-07T20:51:41Z" level=info msg="[TCP] 198.18.0.1:53562 <-> 193.228.196.69:443"
time="2024-04-07T20:51:42Z" level=info msg="[TCP] 198.18.0.1:54060 <-> 104.26.12.49:443"
time="2024-04-07T20:51:43Z" level=info msg="[TCP] 198.18.0.1:40956 <-> 172.67.69.102:443"
time="2024-04-07T20:51:44Z" level=info msg="[TCP] 198.18.0.1:59366 <-> 169.150.247.36:443"
time="2024-04-07T20:51:45Z" level=info msg="[TCP] 198.18.0.1:49010 <-> 104.26.12.245:443"
time="2024-04-07T20:51:46Z" level=info msg="[TCP] 198.18.0.1:54072 <-> 104.26.12.49:443"
time="2024-04-07T20:51:51Z" level=info msg="[TCP] 198.18.0.1:34316 <-> 188.114.96.3:443"
time="2024-04-07T20:51:51Z" level=info msg="[TCP] 198.18.0.1:34328 <-> 188.114.96.3:443"
time="2024-04-07T20:51:51Z" level=info msg="[TCP] 198.18.0.1:52976 <-> 188.114.97.3:443"
time="2024-04-07T20:51:52Z" level=info msg="[TCP] 198.18.0.1:34338 <-> 188.114.96.3:443"
time="2024-04-07T20:51:52Z" level=info msg="[TCP] 198.18.0.1:52990 <-> 188.114.97.3:443"
time="2024-04-07T20:51:53Z" level=info msg="[TCP] 198.18.0.1:36962 <-> 104.26.13.49:443"
time="2024-04-07T20:51:57Z" level=info msg="[TCP] 172.17.0.2:47539 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:52801 <-> 188.165.58.33:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:44871 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:54903 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:45945 <-> 151.101.66.167:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:55285 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:55003 <-> 1.0.0.1:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:55361 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:48539 <-> 140.82.121.4:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:55961 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:56683 <-> 51.195.31.115:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:46689 <-> 104.244.42.193:443"
time="2024-04-07T20:51:59Z" level=info msg="[TCP] 172.17.0.2:42215 <-> 185.125.190.29:443"
time="2024-04-07T20:52:00Z" level=info msg="[TCP] 172.17.0.2:42457 <-> 146.75.122.167:443"
time="2024-04-07T20:52:00Z" level=info msg="[TCP] 172.17.0.2:60981 <-> 51.195.31.115:443"
time="2024-04-07T20:52:01Z" level=info msg="[TCP] 172.17.0.2:58639 <-> 51.195.31.115:443"
time="2024-04-07T20:52:01Z" level=info msg="[TCP] 172.17.0.2:34261 <-> 188.165.60.250:443"
time="2024-04-07T20:52:01Z" level=info msg="[TCP] 172.17.0.2:51053 <-> 188.165.58.33:443"
time="2024-04-07T20:52:01Z" level=info msg="[TCP] 172.17.0.2:49171 <-> 51.195.31.115:443"
time="2024-04-07T20:52:02Z" level=info msg="[TCP] 198.18.0.1:36924 <-> 37.27.114.122:25565"
time="2024-04-07T20:52:04Z" level=info msg="[TCP] 172.17.0.2:53691 <-> 144.76.114.3:80"
time="2024-04-07T20:52:04Z" level=info msg="[TCP] 172.17.0.2:47105 <-> 51.195.31.115:443"
time="2024-04-07T20:52:04Z" level=info msg="[TCP] 172.17.0.2:38671 <-> 52.85.92.85:443"
time="2024-04-07T20:52:04Z" level=info msg="[TCP] 172.17.0.2:51823 <-> 51.195.31.115:443"
time="2024-04-07T20:52:05Z" level=info msg="[TCP] 198.18.0.1:44770 <-> 18.167.66.16:8080"
time="2024-04-07T20:52:05Z" level=info msg="[TCP] 172.17.0.2:56641 <-> 51.195.31.115:443"
time="2024-04-07T20:52:05Z" level=info msg="[TCP] 172.17.0.2:40027 <-> 52.35.137.82:443"
time="2024-04-07T20:52:06Z" level=info msg="[TCP] 198.18.0.1:44784 <-> 18.167.66.16:8080"
time="2024-04-07T20:52:07Z" level=info msg="[TCP] 198.18.0.1:41894 <-> 104.26.13.49:443"
time="2024-04-07T20:52:08Z" level=info msg="[TCP] 198.18.0.1:58522 <-> 104.26.13.245:443"
time="2024-04-07T20:52:11Z" level=info msg="[TCP] 198.18.0.1:41878 <-> 151.101.2.87:443"
time="2024-04-07T20:52:11Z" level=info msg="[TCP] 198.18.0.1:46986 <-> 143.198.246.222:443"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:41882 <-> 151.101.2.87:443"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:52764 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:34810 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:52780 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44814 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:47994 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:41586 <-> 24.144.71.154:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:53268 <-> 104.16.248.249:443"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:48006 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44822 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:47988 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44834 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:45050 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:52784 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:45060 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44846 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:34820 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44850 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:34836 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:44858 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:45066 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:45036 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:52794 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:13Z" level=info msg="[TCP] 198.18.0.1:34840 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34848 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44862 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34864 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44864 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:48022 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44866 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34872 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:52810 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34884 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44870 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:58004 <-> 172.67.184.82:443"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34896 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34902 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:52814 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45076 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44874 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34910 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44876 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45082 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:41596 <-> 24.144.71.154:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44890 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:48038 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44884 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:48050 <-> 24.144.68.195:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44900 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:52834 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45100 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34914 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44902 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:34928 <-> 209.38.174.34:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44904 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45092 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45110 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:44906 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:52838 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:45124 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:14Z" level=info msg="[TCP] 198.18.0.1:52846 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:52820 <-> 24.199.78.244:30944"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:34188 <-> 185.135.81.106:443"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:38176 <-> 142.250.186.34:443"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:44918 <-> 24.144.71.110:30944"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:56770 <-> 80.240.22.34:7072"
time="2024-04-07T20:52:15Z" level=info msg="[TCP] 198.18.0.1:45122 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:16Z" level=info msg="[TCP] 198.18.0.1:59936 <-> 104.16.111.210:443"
time="2024-04-07T20:52:17Z" level=info msg="[TCP] 198.18.0.1:45134 <-> 209.38.172.228:30945"
time="2024-04-07T20:52:17Z" level=info msg="[TCP] 198.18.0.1:33206 <-> 185.135.81.106:51022"
time="2024-04-07T20:52:17Z" level=info msg="[TCP] 198.18.0.1:47152 <-> 52.33.204.55:443"
time="2024-04-07T20:52:17Z" level=info msg="[TCP] 198.18.0.1:35148 <-> 172.67.166.76:80"
time="2024-04-07T20:52:19Z" level=info msg="[TCP] 198.18.0.1:38092 <-> 185.135.81.106:51020"
time="2024-04-07T20:52:30Z" level=info msg="[TCP] 198.18.0.1:36642 <-> 104.16.249.249:443"
time="2024-04-07T20:52:30Z" level=info msg="[TCP] 198.18.0.1:36324 <-> 110.93.147.42:443"
time="2024-04-07T20:52:30Z" level=info msg="[TCP] 198.18.0.1:39884 <-> 104.16.123.96:443"
time="2024-04-07T20:52:30Z" level=info msg="[TCP] 198.18.0.1:55318 <-> 104.16.248.249:443"
```

![image](https://github.com/engageub/InternetIncome/assets/1298861/46aa4bd1-5602-49af-add2-aca8f4471ce8)

Again. Port 6000 is reserved for X11 and blocked by browsers for security reasons. You cannot use that.
Other projects have the same problem.
- https://github.com/ankane/pghero/issues/86
- https://www.reddit.com/r/firefox/comments/ttms50/since_when_was_this_a_thing_in_firefox_trying_to/

**Comment by engageub** [comment]
Is serveo not working as its not a direct access.

**Comment by copystring** [comment]
No. serveo does not work with port 6000.

**Comment by copystring** [comment]
> ![image](https://private-user-images.githubusercontent.com/1298861/320312096-845befbc-3b62-4909-9944-8b05ea852502.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1MjI1MTMsIm5iZiI6MTcxMjUyMjIxMywicGF0aCI6Ii8xMjk4ODYxLzMyMDMxMjA5Ni04NDViZWZiYy0zYjYyLTQ5MDktOTk0NC04YjA1ZWE4NTI1MDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDdUMjAzNjUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGVmYjFlNmM3MmE2YWFiZjQyMTJkZmRjNjk3NjA4MzVhOGMyNTA0YWYyMjE1MWVhYWQzYWQ1YmE4NmNhZmUzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cXTrAH_gsaSpvcn7j74XkWKWNXfoLZZGVsN5xEzA5SI)

**Comment by engageub** [comment]
> > ![image](https://private-user-images.githubusercontent.com/1298861/320312096-845befbc-3b62-4909-9944-8b05ea852502.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1MjI1MTMsIm5iZiI6MTcxMjUyMjIxMywicGF0aCI6Ii8xMjk4ODYxLzMyMDMxMjA5Ni04NDViZWZiYy0zYjYyLTQ5MDktOTk0NC04YjA1ZWE4NTI1MDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDdUMjAzNjUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGVmYjFlNmM3MmE2YWFiZjQyMTJkZmRjNjk3NjA4MzVhOGMyNTA0YWYyMjE1MWVhYWQzYWQ1YmE4NmNhZmUzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cXTrAH_gsaSpvcn7j74XkWKWNXfoLZZGVsN5xEzA5SI)

The image cannot be accessed. Could you please let me know which OS and VPS you are using.

**Comment by copystring** [comment]
No VPS. I have my own server at home with proxmox debian

**Comment by copystring** [comment]
But it has nothing to with the server. The error is from the browser not the server.

**Comment by engageub** [comment]
Can you try the following instead of localhost.

ssh -R 80:0.0.0.0:6000 serveo.net

or

ssh -R 80:127.0.0.1:6000 serveo.net


Thank you

**Comment by copystring** [comment]
> > > ![image](https://private-user-images.githubusercontent.com/1298861/320312096-845befbc-3b62-4909-9944-8b05ea852502.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1MjI1MTMsIm5iZiI6MTcxMjUyMjIxMywicGF0aCI6Ii8xMjk4ODYxLzMyMDMxMjA5Ni04NDViZWZiYy0zYjYyLTQ5MDktOTk0NC04YjA1ZWE4NTI1MDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDdUMjAzNjUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGVmYjFlNmM3MmE2YWFiZjQyMTJkZmRjNjk3NjA4MzVhOGMyNTA0YWYyMjE1MWVhYWQzYWQ1YmE4NmNhZmUzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cXTrAH_gsaSpvcn7j74XkWKWNXfoLZZGVsN5xEzA5SI)
> 
> The image cannot be accessed. Could you please let me know which OS and VPS you are using.

This is from localhost. I have already done this.

**Comment by engageub** [comment]
For proxmox there may be firewall issues. Please find below.

https://forum.proxmox.com/threads/outgoing-ports-blocked-on-lxc.92548/
https://forum.proxmox.com/threads/how-to-forward-ports-for-x11-on-proxmox-to-a-mv.79129/

**Comment by copystring** [comment]
Please. This is not an issue with proxmox or firewall. It has nothing to do with my server. The problem is port 6000. You are not allowed to use this. When I change this to port 8000 or 12000 it works fine.

**Comment by engageub** [comment]
Else, the code will be changed to start with 6001 instead of 6000 or something else.
I am using ubuntu and I don't have any issues with port 6000. 
As mentioned in the above forums users were having issues with port 6000 on proxmox.

Thank you

**Comment by copystring** [comment]
Changing the first port to 6001 is OK.

**Comment by engageub** [comment]
Can you also raise an issue with proxmox to understand why port 6000 is not accessible when the port can be opened?

Thank you

**Comment by copystring** [comment]
> As mentioned in the above forums users were having issues with port 6000 on proxmox.

Port 6000 has nothing to do with proxmox. When I use my phone without WiFi on mobile network and open localhost:6000 I get the same error message `ERR_UNSAFE_PORT` because you are not allowed to use port 6000. This not a problem of my server. No issue with proxmox.
![image](https://github.com/engageub/InternetIncome/assets/1298861/86cbcae8-345d-4058-b8b7-d501bd8bc28c)

`ERR_UNSAFE_PORT` means browsers cannot use this port. You need to change your script to start from a different port. You cannot start from first port 6000.

**Comment by engageub** [comment]
Also on your host, can you run the curl command as I see the output here without any issues. 

curl http://127.0.0.1:6000

**Comment by engageub** [comment]
> > As mentioned in the above forums users were having issues with port 6000 on proxmox.
> 
> Port 6000 has nothing to do with proxmox. When I use my phone without WiFi on mobile network and open localhost:6000 I get the same error message `ERR_UNSAFE_PORT` because you are not allowed to use port 6000. This not a problem of my server. No issue with proxmox. ![image](https://private-user-images.githubusercontent.com/1298861/320314337-86cbcae8-345d-4058-b8b7-d501bd8bc28c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1MjUyMzUsIm5iZiI6MTcxMjUyNDkzNSwicGF0aCI6Ii8xMjk4ODYxLzMyMDMxNDMzNy04NmNiY2FlOC0zNDVkLTQwNTgtYjhiNy1kNTAxYmQ4YmMyOGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDdUMjEyMjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTQ3NmY5ZWU3ZmEzYjdlODk2ZjQ2ZjYxYjUzMmViYjRkM2VmMmIxMzRjYWVhYzVlMmQ1YTAxYjc1N2FkZTUxMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wfPp0RKYLJLjwyzegKEJuGh549JyY_anYqVFXCu2_xQ)
> 
> `ERR_UNSAFE_PORT` means browsers cannot use this port. You need to change your script to start from a different port. You cannot start from first port 6000.

That might be default message for port 6000 but I am able to access it here using serveo.net while you are not.

**Comment by copystring** [comment]
> That might be default message for port 6000 but I am able to access it here using serveo.net while you are not.

Which browser do you use?
Port 6000 is a known blocked port on Firefox and Chrome.
See this list: https://neo4j.com/developer/kb/list-of-restricted-ports-in-browsers/
![image](https://github.com/engageub/InternetIncome/assets/1298861/77efe705-3e21-4f0f-a3ac-b9a9b12ffb49)

**Comment by engageub** [comment]
I used chrome using serveo.net. Please find the output for curl command on port 6000.
Can you share the output from your host



ubuntu@ubuntu:~/customBrowser/InternetIncome-test$ curl http://127.0.0.1:6000
<!DOCTYPE html>
<html lang="en" class="noVNC_loading">
<head>

    

    <title></title>

    <meta charset="utf-8">

    
    <link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png?v=c0d925cce1">
    <link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png?v=c0d925cce1">
    <link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png?v=c0d925cce1">
    <link rel="manifest" href="site.webmanifest?v=c0d925cce1">
    <link rel="mask-icon" href="safari-pinned-tab.svg?v=c0d925cce1" color="#5bbad5">
    <link rel="shortcut icon" href="favicon.ico?v=c0d925cce1">
    <meta name="msapplication-TileColor" content="#2d89ef">
    <meta name="theme-color" content="#ffffff">
    

    
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />

    
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

    
    <link rel="stylesheet" href="app/styles/font-awesome.min.css?v=ca8fe3770f">
    <link rel="stylesheet" href="app/styles/bootstrap-nightshade.min.css?v=ca8fe3770f">
    <link rel="stylesheet" href="app/styles/base.css?v=ca8fe3770f">

    
    <link rel="preload" as="image" href="app/images/info.svg">
    <link rel="preload" as="image" href="app/images/error.svg">
    <link rel="preload" as="image" href="app/images/warning.svg">

    
    <script src="app/bootstrap.bundle.min.js?v=ca8fe3770f"></script>
    <script src="app/error-handler.js?v=ca8fe3770f"></script>
    <script type="module" crossorigin="anonymous" src="app/ui.js?v=ca8fe3770f"></script>
</head>

<body>
    
    <div id="noVNC_fallback_error" class="noVNC_center">
        <div>
            <div>Unexpected error encountered:</div>
            <br>
            <div id="noVNC_fallback_errormsg"></div>
        </div>
    </div>

    
    <div id="noVNC_control_bar_anchor" class="noVNC_vcenter noVNC_hidden">
        <div id="noVNC_control_bar" class="d-flex flex-row">
            
            <div class="card noVNC_scroll" style="width: 21rem; font-size: 14px;">
                
                <div class="card-header d-flex align-items-center">
                    <img class="pe-2" style="height: 25px;" src="app/images/icons/master_icon.png?v=ca8fe3770f" id="noVNC_app_logo">
                    <h5 class="m-0" name="noVNC_app_name">DockerApp</h5>
                    <div class="ms-auto"></div>
                </div>
                
                <ul class="list-group list-group-flush">
                    
                    <li class="list-group-item" id="noVNC_action_icons_section">
                        <div class="d-flex justify-content-around align-items-center">
                            <a class="btn shadow-none p-0" href="#" title="Show Keyboard" id="noVNC_keyboard_button"><i class="fa fa-keyboard-o fa-fw fs-4"></i></a>
                            <a class="btn shadow-none p-0 noVNC_hidden" href="#" title="Fullscreen" id="noVNC_fullscreen_button"><i class="fa fa-arrows-alt fa-fw fs-4"></i></a>
                            <a class="btn shadow-none p-0 noVNC_hidden" href="#" title="Move/Drag Viewport" id="noVNC_view_drag_button"><i class="fa fa-hand-rock-o fa-fw fs-4"></i></a>
                        </div>
                    </li>
                    
                    <li class="list-group-item">
                        
                        <label class="custom-accordion-button text-nowrap" data-bs-toggle="collapse" data-bs-target="#clipboardCollapse">Clipboard</label>
                        <div class="input-group input-group-sm collapse show" id="clipboardCollapse">
                            <textarea class="form-control" wrap="off" rows="3" id="noVNC_clipboard_text"></textarea>
                            <button class="btn btn-outline-secondary" type="button" id="noVNC_clipboard_clear_button">Clear</button>
                        </div>
                    </li>
                    
                    <li class="list-group-item">
                        <label class="custom-accordion-button text-nowrap" data-bs-toggle="collapse" data-bs-target="#settingsCollapse">Settings</label>
                        <div class="collapse show" id="settingsCollapse">
                            
                            <div class="form-check form-switch mb-2">
                                <input class="form-check-input" type="checkbox" role="switch" id="noVNC_setting_view_clip">
                                <label class="form-check-label text-nowrap" for="noVNC_setting_view_clip">Clip to Window</label>
                            </div>
                            
                            <div class="input-group input-group-sm mb-2">
                                <label class="input-group-text" for="noVNC_setting_resize">Scaling Mode</label>
                                <select class="form-select" id="noVNC_setting_resize">
                                    <option value="off">None</option>
                                    <option value="scale">Local Scaling</option>
                                    <option value="remote">Remote Resizing</option>
                                </select>
                            </div>
                            
                            <div>
                                <label for="noVNC_setting_quality" class="text-nowrap">Quality</label>
                                <input type="range" min="0" max="9" value="6" class="form-range" id="noVNC_setting_quality">
                            </div>
                            
                            <div class="mb-1">
                                <label for="noVNC_setting_compression" class="text-nowrap">Compression Level</label>
                                <input type="range" min="0" max="9" value="2" class="form-range" id="noVNC_setting_compression">
                            </div>
                            
                            <div class="input-group input-group-sm">
                                <label class="input-group-text" for="noVNC_setting_logging">Logging</label>
                                <select class="form-select" id="noVNC_setting_logging">
                                    <option value="error">Error</option>
                                    <option value="warn">Warning</option>
                                    <option value="info">Info</option>
                                    <option value="debug">Debug</option>
                                </select>
                            </div>
                        </div>
                    </li>
                </ul>
                
                <div id="noVNC_version_footer" class="card-footer noVNC_hidden">
                    <div id="noVNC_version_app" class="noVNC_hidden"><small class="text-muted text-nowrap">Application vX.Y.Z</small></div>
                    <div id="noVNC_version_docker_image" class="noVNC_hidden"><small class="text-muted text-nowrap">Docker Image vX.Y.Z</small></div>
                </div>
            </div>
            
            <div id="noVNC_control_bar_handle" class="handle">
                <div class="dots"></div>
            </div>
        </div>
        
        <div id="noVNC_control_bar_hint"></div>
    </div> 

    
    <div id="noVNC_status"></div>

    
    <div class="noVNC_center noVNC_connect_layer">
        <div id="noVNC_credentials_dlg">
            <div class="card">
                <div class="card-header" name="noVNC_app_name">DockerApp</div>
                    <div class="card-body">
                        <h5 class="card-title">Credentials needed</h5>
                        <p class="card-text">Please enter the credentials to connect.</p>
                        <form>
                            <div id="noVNC_username_block" mclass="mb-3">
                                <input type="text" class="form-control" placeholder="Username" id="noVNC_username_input">
                            </div>
                            <div id="noVNC_password_block" class="mb-3">
                                <input type="password" class="form-control" placeholder="Password" id="noVNC_password_input">
                            </div>
                            <button type="submit" id="noVNC_credentials_button" class="btn btn-primary">Submit</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    </div>

    
    <div id="noVNC_transition">
        <div id="noVNC_transition_text"></div>
        <div class="noVNC_spinner"></div>
    </div>

    
    <div id="noVNC_container">
        
        <textarea id="noVNC_keyboardinput" autocapitalize="off"
            autocomplete="off" spellcheck="false" tabindex="-1"></textarea>
    </div>

</body>
</html>

**Comment by copystring** [comment]
> Also on your host, can you run the curl command as I see the output here without any issues.
> 
> ```
> curl http://127.0.0.1:6000
> ```

![image](https://github.com/engageub/InternetIncome/assets/1298861/37018cf1-24cd-4c31-b1f5-c841483c6582)

Curl works fine. Firefox and Chrome not. Seem my comment above.

**Comment by engageub** [comment]
Can you use the same url instead of localhost in browser and serveo.net

**Comment by copystring** [comment]
Posteo works
![image](https://github.com/engageub/InternetIncome/assets/1298861/b7a6c36b-52bc-470f-9df1-7663a7d5b2d9)

**Comment by engageub** [comment]
okay. I mentioned earlier to use 0.0.0.0 or 127.0.0.1 . If its working, no changes are required. 

Thank you

**Comment by copystring** [comment]
OK. It works with serveo.net, but I think you should still change the port for security reasons, as it's blocked by browsers.

**Comment by copystring** [comment]
![image](https://github.com/engageub/InternetIncome/assets/1298861/1268c1e7-4707-4e41-8dee-d4a5dd686646)

**Comment by copystring** [comment]
Anyone using X11 will not be able to access the first port of custom Firefox.

**Comment by engageub** [comment]
Custom first port has been changed to 5000.
Please note that port assignment is done based on availability. The script checks if port is opened or not.
If X11 is running, it will not assign that port. So there is no issue with the script. Only GUI is not accessible directly instead through serveo.net. 

If you wish to access unsafe ports in chrome, you will have to change the following.

https://superuser.com/questions/188006/how-to-fix-err-unsafe-port-error-on-chrome-when-browsing-to-unsafe-ports

Thank you

**Comment by copystring** [comment]
Thank you.

> If you wish to access unsafe ports in chrome, you will have to change the following.

Yes, I've read this, but I don't want to allow this, but no issues anymore with first port as 5000.
